source/extensions/quic_listeners/quiche/envoy_quic_utils.h

```diff
@@ -61,7 +61,10 @@ std::unique_ptr<T> spdyHeaderBlockToEnvoyHeaders(const spdy::SpdyHeaderBlock& he
   for (auto entry : header_block) {
     // TODO(danzh): Avoid temporary strings and addCopy() with string_view.
     std::string key(entry.first);
-    headers->addCopy(Http::LowerCaseString(key), entry.second);
+    std::vector<absl::string_view> values = absl::StrSplit(entry.second, '\0');
```

Comment:
I was surprised to see splitting on `\0` and so I went looking around and found the definition in spdy/core/spdy_header_block.h which confirms it, but not all the time:
```
  // If a header with the key is already present, then append the value to the
  // existing header value, NUL ("\0") separated unless the key is cookie, in
  // which case the separator is "; ".
  // If there is no such key, a new header with the key and value is added.
  void AppendValueOrAddHeader(const absl::string_view key,
                              const absl::string_view value);
```
I've never worked with that library before but I wonder if it supplies some sort of split API we should be using rather than calling StrSplit directly.

Reply:
I was mostly just thinking that the \0 delim looks like an implementation detail of spdy_header_block.h and the detail should ideally be abstracted away at that level.

And yes I was also wondering specifically about what the behavior would be with cookies.

Reply:
Unfortunately, SpdyHeaderBlock doesn't have such API to split the `\0` delimited header values. Even quic qpack code has to write its own logic to handle this special case.

Reply:
Some comments here might help at least clear up the mysticism, if it's really impossible to create an abstraction boundary around this.

Reply:
+1 for more comments

Reply:
done